packages/language/src/preprocessor/instruction-generator.ts

```diff
@@ -336,6 +336,28 @@ function generateDoInstruction(
       while: whileCond,
     };
   }
+  if (node.doType3) {
+    const variable = generateReferenceItemInstruction(node.doType3.variable!);
```

Comment:
Minor: don't coerce possible null values, simply return `undefined`.
```suggestion
    if (!node.doType3.variable) {
      return undefined;
    }
    const variable = generateReferenceItemInstruction(node.doType3.variable);
```